R/parse_system.R

```diff
@@ -178,6 +178,36 @@ parse_system_overall <- function(exprs, call) {
        exprs = exprs)
 }
 
+parse_array_range_depends <- function(x) {
```

Comment:
All this code is in the wrong place; you can and should be doing this in parse_expr, similarly to how we do depenencies for rhs (basically this file is for bits of the parse that depend on *multiple* equations at once, but what you are pulling apart here is just the single equation)

Reply:
Code to do this looks approximately like:

```diff
+++ b/R/parse_expr.R
@@ -165,15 +165,20 @@ parse_expr_assignment_lhs <- function(lhs, src, call) {
                  seq_len(length(lhs) - 2),
                  lhs[-(1:2)],
                  MoreArgs = list(name = name, src = src, call = call))
+    depends <- join_dependencies(
+      lapply(array, function(x)
+        join_dependencies(lapply(x[c("at", "from", "to")], find_dependencies))))
   } else {
     name <- parse_expr_check_lhs_name(lhs, special, is_array, src, call)
     array <- NULL
+    depends <- NULL
   }
```

(this is a patch against your branch so you can apply it directly).

Then in `parse_expr_assignment` join the `lhs` and `rhs` dependencies together (again using `join_dependencies` and see if that gets you close.  Then delete all this code here.